### PR TITLE
Allow multiple users to import the same scenario

### PR DESCRIPTION
**Pull request by TildeWill**
- also renames should_import? to import_confirmed? for readability

Related issue: #968

### CODE REVIEW

File: spec/models/scenario_import_spec.rb [hunk 1]
```diff
@@ -507,6 +507,31 @@
...
+        _existing_scenerio.guid = guid
+        _existing_scenerio.save!
+        _existing_scenerio
+      end
+      
+      describe "#import" do
+        it "makes a new scenario" do
+          expect {
+            scenario_import.import(:skip_agents => true)
+          }.to change { users(:bob).scenarios.count }.by(1)
```
Comment (cantino) [inline spec/models/scenario_import_spec.rb]: I'd also assert that it doesn't touch Jane's scenario in any way.

File: app/controllers/scenario_imports_controller.rb [hunk 2]
```diff
@@ -7,11 +7,7 @@ def create
     @scenario_import = ScenarioImport.new(params[:scenario_import])
     @scenario_import.set_user(current_user)
 
-    if @scenario_import.will_request_local?(scenarios_url)
-      render :text => 'Sorry, you cannot import a Scenario by URL from your own Huginn server.' and return
```
Comment (cantino) [inline app/controllers/scenario_imports_controller.rb]: I wish I could remember why I added this restriction...

Comment (TildeWill) [inline app/controllers/scenario_imports_controller.rb]: I don't know the history, but If the guids at one time were expected to be unique across an entire instance of Huginn, then it makes perfect sense.

File: spec/models/scenario_import_spec.rb (hunk 1 above)
Comment (cantino) [inline spec/models/scenario_import_spec.rb]: I'd also assert that now `Agent.where(guid: guid).count` is 2

File: app/controllers/scenario_imports_controller.rb (hunk 2 above)
Comment (cantino) [inline app/controllers/scenario_imports_controller.rb]: I think you're right, but now it looks like we always scope the finders to the user and have `add_index :scenarios, [:user_id, :guid], :unique => true`, so it should be okay, I agree.

**Comment by cantino** [comment]
Is this done?

**Comment by TildeWill** [comment]
Yes, ready for review.

**Comment by cantino** [comment]
Looks great, thanks @TildeWill!

**Pull request merged**